### DISCUSSION TITLE
Option --http-relative-path not available in prod mode

### DISCUSSION THREAD

**Discussion by devopstom-user** [keycloak/keycloak#10567] [Archive]
Hi,

migration guide states the removal of /auth from context path and that it could be set if need be for compatibility via the option `--http-relative-path`. However it looks like this is only available in dev mode, in prod mode getting this error during startup:

```kubectl logs -f test-keycloak-1
Unknown option: '--http-relative-path=/auth'
Possible solutions: --https-certificate-file, --https-cipher-suites, --https-port, --http-port, --https-key-store-type, --http-enabled, --https-client-auth, --http-host, --https-protocols, --https-trust-store-file, --https-trust-store-password, --https-trust-store-type, --https-certificate-key-file, --https-key-store-file, --https-key-store-password
Try 'kc.sh start --help' for more information on the available options.
```
not sure if this is a bug or how else to run this with this "/auth" context path.
Any hints much appreciated.

**Comment by Wolfsrudel** [comment]
It's a build option, so you'll have to use it in this step (https://www.keycloak.org/server/all-config?f=build).
```
kc.sh start --auto-build --http-relative-path=/auth
```
or
```
kc.sh build --http-relative-path=/auth
kc.sh start
```

**Reply by devopstom-user** [reply]
ah right, I used it in start-dev mode as an cmd line option and it worked, so was confused a bit. Indeed it works the way you describe and also when added into Dockerfile and re-building the image.
Thanks!

**Reply by folamig** [reply]
were you able to start it in production mode

**Comment by sreekesh93** [comment]
This is my docker file

FROM quay.io/keycloak/keycloak:17.0.0
ENTRYPOINT ["/opt/keycloak/bin/kc.sh", "start","--auto-build","--http-relative-path=/auth"]  

But still this path is not added
Note:have supplied env variables separately

**Reply by devopstom-user** [reply]
@sreekesh93  have you tried to set it as as ENV vars in Docker file and then build it?

ENV KC_METRICS_ENABLED=true
ENV KC_HTTP_RELATIVE_PATH="/auth"

RUN /opt/keycloak/bin/kc.sh build

I followed this guide: https://www.keycloak.org/server/containers

**Comment by mehtadev17** [comment]
+1 on 20.0.2
